src/sagas/walletConnect.js

```diff
@@ -0,0 +1,494 @@
+
+  const activeSessions = yield call(() => web3wallet.getActiveSessions());
+  yield put(setWalletConnectSessions(activeSessions));
+}
+
+/**
+ * @param {Web3Wallet} web3wallet The WalletConnect web3wallet instance
+ */
+export function* setupListeners(web3wallet) {
+  const channel = eventChannel((emitter) => {
+    const listener = (state) => emitter(state);
+
+    web3wallet.on('session_request', (event) => {
+      emitter({
+        type: 'WC_SESSION_REQUEST',
+        data: event,
+      });
+    });
+
+    web3wallet.on('session_proposal', async (proposal) => {
+      emitter({
+        type: 'WC_SESSION_PROPOSAL',
+        data: proposal,
+      });
+    });
+
+    web3wallet.on('session_delete', async (session) => {
+      emitter({
+        type: 'WC_SESSION_DELETE',
+        data: session,
+      });
+    });
+
+    return () => {
+      web3wallet.removeListener('session_request', listener);
+      web3wallet.removeListener('session_proposal', listener);
+      web3wallet.removeListener('session_delete', listener);
+    };
+  });
+
+  try {
+    while (true) {
+      const message = yield take(channel);
+
+      yield put({
+        type: message.type,
+        payload: message.data,
+      });
+    }
+  } finally {
+    if (yield cancelled()) {
+      // When we close the channel, it will remove the event listener
+      channel.close();
+    }
+  }
+}
+
+/**
+ * This saga will publish on the cloud server a RPC message disconnecting
+ * the current client.
+ */
+export function* clearSessions() {
+  const { web3wallet } = yield select((state) => state.walletConnect.client);
+  const activeSessions = yield call(() => web3wallet.getActiveSessions());
+
+  for (const key of Object.keys(activeSessions)) {
+    yield call(() => web3wallet.disconnectSession({
+      topic: activeSessions[key].topic,
+      reason: {
+        code: ERROR_CODES.USER_DISCONNECTED,
+        message: '',
+      },
+    }));
+  }
+
+  yield call(refreshActiveSessions);
+}
+
+/**
+ * This saga will be called (dispatched from the event listener) when a session
+ * is requested from a dApp
+ */
+export function* onSessionRequest(action) {
+  const { payload } = action;
+  const { params } = payload;
+
+  const { web3wallet } = yield select((state) => state.walletConnect.client);
+  const activeSessions = yield call(() => web3wallet.getActiveSessions());
+  const requestSession = activeSessions[payload.topic];
+  if (!requestSession) {
+    console.error('Could not identify the request session, ignoring request..');
+    return;
+  }
+
+  const data = {
+    icon: get(requestSession.peer, 'metadata.icons[0]', null),
+    proposer: get(requestSession.peer, 'metadata.name', ''),
+    url: get(requestSession.peer, 'metadata.url', ''),
+    description: get(requestSession.peer, 'metadata.description', ''),
+  };
+
+  switch (params.request.method) {
+    case AVAILABLE_METHODS.HATHOR_SIGN_MESSAGE:
```

Comment:
So this session started by the dapp is not the session start, it's actually a method call? I thought this would just start the first iteration between the wallet and the dapp.

Reply:
The handling of the initial iteration between the wallet and the dapp (through the QRCode) is done through the `onSessionProposal` saga

Every "session" request is done through the `onSessionRequest` saga

The request might be anything defined on `AVAILABLE_METHODS` which in this PR is only the `signMessage` request

Reply:
Can we add this explanation in the methods, please? It's not so intuitive, in my opinion

Reply:
Yes, done in e634d2e04631972c316c060df4d659a375b37d22